paddle/fluid/framework/details/scope_buffered_ssa_graph_executor.cc

```diff
@@ -64,21 +64,30 @@ FeedFetchList ScopeBufferedSSAGraphExecutor::Run(
   }
 
   platform::RecordEvent e("ScopeBufferedSSAGraphExecutorAfterRun", nullptr);
-  drop_scope_counter_ += 1;
+  ++drop_scope_counter_;
 
-  if (!fetch_tensors.empty() ||
-      drop_scope_counter_ == strategy_.num_iteration_per_drop_scope_) {
-    drop_scope_counter_ = 0;
+  if (!fetch_tensors.empty()) {
     // Wait All computational streams
     for (auto p : places_) {
       platform::DeviceContextPool::Instance().Get(p)->Wait();
     }
+  }
+
+  if (drop_scope_counter_ == strategy_.num_iteration_per_drop_scope_) {
```

Comment:
@chengduoZH @sneaxiy any comment about this change?

Reply:
It looks ok.